Rematch/UI.lua

```diff
@@ -49,77 +46,50 @@ local scriptButtons = setmetatable({}, {
-        end
+function RematchPlugin:SetupUI()
+    local rematchVersion = ns.Version:Current('Rematch')
 
-        self:RemoveScript(old)
-        self:AddScript(new, script)
+    -- Add menu to edit script
+    if rematchVersion >= ns.Version:New(5, 0, 0, 0) then
+        local afterText = Rematch.localization['Set Notes'] -- Use Rematch's locale string.
+        Rematch.menus:AddToMenu('TeamMenu', scriptMenuItem, afterText)
+        Rematch.menus:AddToMenu('LoadedTeamMenu', scriptMenuItem, afterText)
+    else
+        tinsert(Rematch:GetMenu('TeamMenu'), 6, scriptMenuItem)
     end
 
-    self:RawHook(Rematch, 'SaveAsAccept', function(...)
-        safecall(function()
-            local team, key = Rematch:GetSideline()
-            if not RematchSaved[key] or not Rematch:SidelinePetsDifferentThan(key) then
-                rename(Rematch:GetSidelineContext('originalKey'), key)
+    -- When a script is added/removed, refresh the teams list.
+    self.updateFrames = nil
+    if rematchVersion >= ns.Version:New(5, 0, 0, 0) then
+        self.updateFrames = function()
+            if Rematch.frame:IsVisible() then
+                Rematch.frame:Update()
             end
-        end)
-        return self.hooks[Rematch].SaveAsAccept(...)
-    end, true)
-
-    self:SecureHook(Rematch, 'OverwriteAccept', function()
-        safecall(function()
-            rename(Rematch:GetSidelineContext('originalKey'), select(2, Rematch:GetSideline()))
-        end)
-    end)
-
-    -- team update
-
-    self:RegisterMessage('PET_BATTLE_SCRIPT_SCRIPT_LIST_UPDATE', function()
-        if RematchLoadedTeamPanel:IsVisible() then
-            RematchLoadedTeamPanel:Update()
         end
-        if RematchTeamPanel:IsVisible() then
-            if RematchTeamPanel.UpdateList then
-                RematchTeamPanel:UpdateList()
-            elseif RematchTeamPanel.List then
-                RematchTeamPanel.List:Update()
+    else
+        self.updateFrames = function()
+            if RematchLoadedTeamPanel:IsVisible() then
+                RematchLoadedTeamPanel:Update()
+            end
+            if RematchTeamPanel:IsVisible() then
+                if RematchTeamPanel.UpdateList then
+                    RematchTeamPanel:UpdateList()
+                elseif RematchTeamPanel.List then
+                    RematchTeamPanel.List:Update()
+                end
             end
         end
-    end)
-
-    local version = ns.Version:Current()
-
-    if version >= ns.Version:New(4, 8, 10, 5) then
-
+    end
+    self:RegisterMessage('PET_BATTLE_SCRIPT_SCRIPT_LIST_UPDATE', self.updateFrames)
+
+    -- Button to indicate a script exists
+    if rematchVersion >= ns.Version:New(5, 0, 0, 0) then
+        -- TODO: restore tooltip/button once Rematch supports that again
+        Rematch.badges:RegisterBadge('teams', 'PetBattleScripts', scriptButtonIcon, nil, function(teamID)
+            -- TODO: No need to `self:IsEnabled() and` if we properly remove on TeardownUI().
```

Comment:
Nice to have [if Gello implements it](https://www.warcraftpets.com/community/forum/viewtopic.php?p=149582#p149582).

Reply:
Remove this TODO and make an issue for tracking it instead?